### ISSUE TITLE
Institutional Office 365 (EAS) Sync Failure, 503 error

### ISSUE THREAD

**Issue by theckhd** [jobisoft/TbSync#125]
## Your environment

TbSync version: 0.7.15.4
Thunderbird version: 52.3.0 (32-bit)

[x] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior
Connect to Office365 and sync calendar, tasks, etc.

## Actual behavior
TBSync either says the server does not provide info about supported ActiveSync versions (if "best available" is chosen), or gives a 503 error (if either v2.5 or v14.0 are chosen).

## Steps to reproduce
Install add-on, add account (configuring manually), attempt to sync. 
TBSync was working wonderfully until about a week or two ago, when it stopped syncing properly. For some reason it failed on my home system first, and was still active on my work machine until Friday afternoon. This morning I tried both the release and beta version of 0.7.15, neither seems to work.
I followed the instructions you gave in [this issue](https://github.com/jobisoft/TbSync/issues/56) (remove account, enable debug, restart Tbird, add account, etc.) to generate a debug log. Sending to you shortly. 
My best guess is that it's related to the fact that it's an institutional account. I have to have two-factor authentication enabled, so when logging into office365 through a browser I need to go through our institution's portal first. That's never been an issue with TBSync before this past week though.
Happy to do any other testing you might need to identify the issue.

**Comment by jobisoft** [comment]
Do you know, if TbSync updated itself between the working and non-working state? The current version 0.7.15 was released by Mozilla on 31.08.

503 means "service not avail", could be, that your IT is no longer allowing TbSync. Have you contacted them about this?

**Comment by jobisoft** [comment]
can you retry with this (rename to xpi) ?
TbSync-beta-0.7.15.5.zip

I cannot see the 503 in the logs. Where did you get that?

What I do not understand: You did not get the "server does not provide info about supported ActiveSync versions" before? That means your server no longer answers the OPTIONS request, that is however before EAS specific user access level restrictions apply. Are you sure, you are still connecting to the correct server?

**Comment by jobisoft** [comment]
upd. link in prev. post.

**Comment by theckhd** [comment]
I don't think TBSync updated itself, but I'm not 100% sure. 
I haven't contacted our IT because I assumed this was managed directly by Microsoft, but I can see if they have any input. Unfortunately their support for Tbird with Office365 has been pretty weak.

I'm was getting the 503 from the "Sync Status" report field in the TbSync account manager. 

Initially, TbSync worked fine - I didn't get the "server does not provide info" message up until a week or two ago. I'm still connecting to the same server (outlook.office365.com), and I doubt that's changed, but I suppose it's possible. They expect us all to use the O365 web interface, so they don't always announce when they change something that would affect the few Tbird users.

Using your new xpi, I repeated the procedure, deleted the account, and created a new one. Now using the "best available" option produces the error message:
"Server does not support ActiveSync v2.5 or v14.0 (only ). TbSync will not work with this ActiveSync server." Trying either option (2.5 or 14.0) individually gives a similar message.

So, I'm guessing I need to contact IT about this? It seems like this is entirely on our end, and not an issue with TbSync.

**Comment by jobisoft** [comment]
I think I found my mistake. Can you retry with this one?
[TbSync-beta.zip](https://github.com/jobisoft/TbSync/files/2419919/TbSync-beta.zip)

**Comment by theckhd** [comment]
Using that version, I get the same error messages as the release ("server does not provide information" for best available, "Communications error (HTTP status 503)" for v2.5 or v14.0 individually). See attached screenshot.

I'll send you the json files and the debug log from the test above.
![tbsync](https://user-images.githubusercontent.com/6124772/46084665-3128f380-c172-11e8-9cb3-d338eda137d0.PNG)

**Comment by jobisoft** [comment]
Good, than this part is fixed. TbSync allows you to use 2.5 or 14.0 even if TbSync did not receive a list of valid versions. However the next request (FolderSync) Returns a 503, which means, Service is not avail.

I have the Feeling, that your IT has disabled EAS. Even though you connect to outlook.office365.com your account is part of some O365 Enterprise account with ist own admin, probably someone from your IT.

It is now time to ask your IT if ActiveSync has been disabled.

Are you on Windows 10? You could than try to use the build in Mail App to connect via Active Sync. If you add an account there, you have to select the last "Advanced Setup" and not the first or second "Exchange / Outlook Option".

**Comment by jobisoft** [comment]
I have two more things for you to try.

1. Microsoft offers a connectivity validation tool, are you able to connect to your server using this webtool?
https://testconnectivity.microsoft.com/

2. I fixed a general exchange bug, which *might* be related to your issue. Can you try the latest beta?
https://github.com/jobisoft/TbSync/releases/download/v0.7.15/TbSync-beta.xpi

**Comment by jobisoft** [comment]
There was again a bug in the asversion code. Fixed in beta .6
Sorry.

The link in the last comment is now pointing to beta .6.

**Comment by theckhd** [comment]
I have Win7 at work, but I can test with the mail app on my Win10 box at home. I've opened a ticket with IT as well, so we'll see where that goes.

**Comment by jobisoft** [comment]
That info about mail app is outdated, please look at this post:
https://github.com/jobisoft/TbSync/issues/125#issuecomment-424990513

Also, if your IT disabled ActiveSync because of TbSync (it did not identify itself sending a DeviceInformation), this is fixed now. Even though this is optional the Exchange Serve 2016 had issues if it was missing.

**Comment by theckhd** [comment]
I didn't get a chance to test Win10 yet anyhow, but beta .6 seems to have fixed the problem. I repeated the same steps (uninstall old TbSync, restart TB, install new TbSync, restart TB, test old account, delete old account, add new one). 

The test sync (new beta, but existing account) failed with the same error message. After deleting that account and creating a fresh one, it synced properly on the first try using "best available."

I haven't seen any response on my IT ticket, so I don't know if they changed something on their end, but I suspect not based on the results of my test sync.

**Comment by jobisoft** [comment]
Ok, that is interesting. That means you have been affected by the general Exchange bug I fixed in that beta. Could you update to the latest stable (released today)
https://github.com/jobisoft/TbSync/releases/download/v0.7.16/TbSync-0.7.16.xpi

and try again, to see if this still works for you? If so, we can close this and you could even withraw the ticket at your IT

**Comment by theckhd** [comment]
Tried this at home, curiously after installing 0.7.16 and trying "best available" it still gave me the error message about the server not providing information about ActiveSync. But when I set it to v2.5 manually, it synced on the first try.

I haven't deleted the account and tried again to figure out why "best available" didn't work (it's not clear to me how, or if, you can edit the settings for an account once it's set up), but if you'd like me to do more testing I can make another one and report the results.

**Comment by jobisoft** [comment]
I would like to investigate the other issue as well, if you have some time for that.

Just to make sure: You no longer have a sync issue and you do not get the 503 anymore. The only remaining problem is that sometimes TbSync reports, that your server did not send any information about the supported versions?

Furthermore: You cannot change options of a connected / enabled account. You can view the current options by double clicking the small slider below the folders list. To edit the settings, you have to disable the account, either by double clicking the entry in the accounts list or by right clicking the entry in the accounts lists and selecting "disable" from the context menu or from the global "account actions" menu.

After looking at the code, it is possible that old accounts (which have not been created with 0.7.16), the wrong message could still popup, but for new accounts, which have been created with 0.7.16 or later, this message should no longer pop up. This is of course only valid, if your server is indeed always sending the data. Can you verify that?

**Comment by theckhd** [comment]
No, I haven't gotten a 503 at all with 0.7.16. And I don't have a sync issue with the account I created this morning. 
I just created a new test account, and it worked the first time with "best available." I also disabled the account I made this morning and changed the setting back to "best available," and that's working now as well. 
My best guess is that I deleted the old account and created the new account without restarting TB, and somehow some of the old settings persisted?
